### PR TITLE
Repo Reorgnization Phase 3: Kotlin Sources

### PR DESCRIPTION
**Pull request by alxrsngrtn**
Reorganizing project Kotlin multiplatform code into new repo structure. 
Creating the wasm and jvm sdk platforms.

### CODE REVIEW

File: javatests/arcs/sdk/kotlin/wasm/Test.kt [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+ * http://polymer.github.io/LICENSE.txt
+ *
+ * Code distributed by Google as part of this project is also subject to an additional IP rights
+ * grant found at
+ * http://polymer.github.io/PATENTS.txt
+ */
+
+package arcs.sdk.kotlin.wasm
+
+annotation class Test
```
Comment (csilvestrini) [inline javatests/arcs/sdk/kotlin/wasm/Test.kt]: Can you add documentation for this? I don't know what it is or why it's here.

**Comment by jasonwyatt** [comment]
In the reorg document, I think @csilvestrini agreed that it would make sense to drop the `/kotlin` part of the new path, since everything is kotlin in there already.

**Comment by csilvestrini** [comment]
> In the reorg design, I think @csilvestrini agreed that it would make sense to drop the `/kotlin` part of the new path, since everything is kotlin in there already.

Yep, I agree. `java/arcs/sdk/{wasm,jvm,android}` would be the final paths.

**Comment by alxrsngrtn** [comment]
> Yep, I agree. `java/arcs/sdk/{wasm,jvm,android}` would be the final paths.

I decided to implement this in the first pass instead of over two. I'm putting shared code in a `common` directory Here's an open question: 
Should the primary package be `arcs.sdk.common` or `arcs.sdk`?

**Comment by jasonwyatt** [comment]
I would vote for arcs.sdk, and reserve `common` for stuff like utilities, with the `arcs.sdk` holding the really important things (like Entity, Particle, etc base classes)

**Comment by alxrsngrtn** [comment]
Ok. I just push a change that flattens the structure (the thing you commented above about). I'll move things to the `arcs.sdk` package now.

**Pull request merged**